src/transactions/feed/detailContent/SwapContent.tsx

```diff
@@ -4,90 +4,122 @@ import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
 import RowDivider from 'src/components/RowDivider'
 import TokenDisplay, { formatValueToDisplay } from 'src/components/TokenDisplay'
+import { NETWORK_NAMES } from 'src/shared/conts'
 import Colors from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
 import { useTokensList } from 'src/tokens/hooks'
 import FeeRowItem from 'src/transactions/feed/detailContent/FeeRowItem'
-import { FeeType, TokenExchange } from 'src/transactions/types'
+import { FeeType, TokenExchange, TransactionStatus } from 'src/transactions/types'
 export interface Props {
-  exchange: TokenExchange
+  transaction: TokenExchange
 }
 
 // Note that this is tested from TransactionDetailsScreen.test.tsx
-export default function SwapContent({ exchange }: Props) {
+export default function SwapContent({ transaction }: Props) {
   const { t } = useTranslation()
   const tokensList = useTokensList()
 
-  const fromTokenSymbol = tokensList.find(
-    (token) => token.tokenId === exchange.outAmount.tokenId
-  )?.symbol
-  const toTokenSymbol = tokensList.find(
-    (token) => token.tokenId === exchange.inAmount.tokenId
-  )?.symbol
+  const fromToken = tokensList.find((token) => token.tokenId === transaction.outAmount.tokenId)
+  const toToken = tokensList.find((token) => token.tokenId === transaction.inAmount.tokenId)
+  const isCrossChainSwap = transaction.__typename === 'CrossChainTokenExchange'
+
+  const showExchangeRate =
+    transaction.status === TransactionStatus.Complete &&
+    !new BigNumber(transaction.inAmount.value).isNaN() &&
+    !!fromToken &&
+    !!toToken
 
   return (
     <View style={styles.contentContainer}>
-      <View style={[styles.row, { paddingBottom: Spacing.Smallest8 }]}>
-        <Text style={styles.bodyText}>{t('swapTransactionDetailPage.swapTo')}</Text>
+      <View style={styles.row}>
+        <Text style={styles.bodyText}>{t('swapTransactionDetailPage.swapFrom')}</Text>
```

Comment:
i assume that switching the from/to ordering for _all_ types of swaps (not just cross-chain) is intended here? (i do see this in the cross-chain designs, but just curious)

Reply:
yes i think so!